app/components/EmbedMarketModal.tsx

```diff
@@ -0,0 +1,168 @@
+}
+
+export const EmbedMarketModal = ({ id }: EmbedMarketModalProps) => {
+  const { isModalOpen, closeModal } = useModal();
+  const [clipboardIcon, setClipboardIcon] = useState<'copy' | 'tick'>('copy');
+  const [isThemePopoverOpen, setThemePopoverOpen] = useState(false);
+  const [selectedTheme, setSelectTheme] = useState(systemTheme);
+
+  const close = () => {
+    closeModal(ModalId.EMBED_MARKET);
+  };
+
+  const IFRAME_WIDTH = 384;
+  const IFRAME_HEIGHT = 222;
+
+  const embedUrl = useMemo(
+    () => `${APP_URL}/embed/market?id=${id}&theme=${selectedTheme.value}`,
+    [id, selectedTheme.value]
+  );
+
+  const getIframeCode = (
+    url: string,
+    width: number = IFRAME_WIDTH,
+    height: number = IFRAME_HEIGHT
+  ) =>
+    `<iframe
+  title="presagio-market-iframe"
+  src=${url} 
+  width=${width} 
+  height=${height} 
+/>`;
+
+  const iframeCode = useMemo(
+    () => getIframeCode(embedUrl, IFRAME_WIDTH, IFRAME_HEIGHT),
+    [embedUrl]
+  );
+
+  const iFrameCodeRender = useMemo(() => {
+    const width = isMobile ? 325 : IFRAME_WIDTH;
+
+    return (
+      <div
+        dangerouslySetInnerHTML={{
+          __html: getIframeCode(embedUrl, width, IFRAME_HEIGHT),
+        }}
+      />
+    );
+  }, [embedUrl]);
+
+  return (
+    <Dialog open={isModalOpen(ModalId.EMBED_MARKET)} onOpenChange={close}>
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle>Embed Market</DialogTitle>
+        </DialogHeader>
+        <DialogBody className="space-y-6 px-3 pb-8 md:px-6">
+          <div className="space-y-4">
+            <h3 className="font-bold">Embed Code</h3>
+            <div className="relative">
+              <div className="absolute right-2 top-2 z-10">
+                <IconButton
+                  onClick={() => {
+                    navigator.clipboard.writeText(iframeCode);
+                    setClipboardIcon('tick');
+
+                    setTimeout(() => {
+                      setClipboardIcon('copy');
+                    }, 3000);
+                  }}
+                  variant="pastel"
+                  name={clipboardIcon}
+                  size="sm"
+                />
+              </div>
+              <SyntaxHighlighter
+                language="html"
+                style={atomDark}
+                customStyle={{ borderRadius: '16px' }}
+              >
+                {iframeCode}
+              </SyntaxHighlighter>
+            </div>
+          </div>
+          <div className="space-y-4">
+            <div className="flex items-center justify-between">
+              <h3 className="font-bold">Preview</h3>
+              <div className="flex items-center space-x-2">
+                <p>Select theme:</p>
+                <div className="w-28">
+                  <Popover open={isThemePopoverOpen} onOpenChange={setThemePopoverOpen}>
+                    <PopoverTrigger asChild>
+                      <Button variant="pastel" className="w-full space-x-2 text-nowrap">
```

Comment:
```suggestion
                      <Button variant="pastel" className="w-full justify-between text-nowrap">
```

I think this works better